controllers/secretagentconfiguration_controller.go

```diff
@@ -67,6 +64,9 @@ func (reconciler *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request
 	)
 
 	var instance v1alpha1.SecretAgentConfiguration
+	var sm *secretsmanager.SecretManager
+	var err error
+
 	if err := reconciler.Get(ctx, req.NamespacedName, &instance); err != nil {
```

Comment:
what's wrong with?

```
err := reconciler.Get
if err != nil {
// thing
}
```

Reply:
Its because of this line: 
if ctx, sm, err = secretsmanager.NewSecretManager(ctx, &instance, cloudCredNS, rClient); err != nil {

Because of the 3 return values, it needs to be declared before.

Reply:
I see... that's happening because 70 is an inline if, and the `err` is scoped  to the if statement.  If you use the "long" way it will be defined and it doesn't have to be declared.

Reply:
https://play.golang.org/p/bnPC5s_Bxwq